crates/llms/src/chat/mistral.rs

```diff
@@ -25,13 +27,16 @@ use mistralrs::{
 };
 use mistralrs_core::{LocalModelPaths, ModelPaths, Pipeline};
 use snafu::ResultExt;
-use std::{path::Path, str::FromStr, sync::Arc};
-use tokio::sync::mpsc::{channel, Receiver, Sender};
+use std::{path::Path, pin::Pin, str::FromStr, sync::Arc};
+use tokio::sync::{
+    mpsc::{channel, Receiver, Sender},
+    RwLock,
+};
 
 pub struct MistralLlama {
     pipeline: Arc<MistralRs>,
     tx: Sender<MistralRsponse>,
-    rx: Receiver<MistralRsponse>,
+    rx: Arc<RwLock<Receiver<MistralRsponse>>>,
```

Comment:
Must add Arc to handle `stream!`. Must add `RwLock` because `stream!` needs mutable access.